libp2p/switch.nim

```diff
@@ -478,6 +510,14 @@ proc stop*(s: Switch) {.async.} =
     except CatchableError as exc:
       warn "error cleaning up transports", msg = exc.msg
 
+  let stopped = await allFuturesThrowing(s.acceptFuts)
```

Comment:
so.. the hope is that none of the accept futures raises? because if any of them did raise, we'll not finish the stop logic here?

Reply:
If it raises it will propagate up to the caller in which case the caller and the caller will have to decide what can be done about it, but we can attempt to handle it here as well given that there probably isn't much else to do.

Reply:
the point is more that the caller _cannot_ reason about what to do if `stop` fails, typically - you can't describe what actions are appropriate because the information is internal to the switch module - specially after stop _partially_ has succeeded (some of the futures may have completed and some not) - this is often the case with these kinds of exceptions that randomly bubble up from composite operations - this is also why raising an exception in a for loop rarely makes sense: there's no way for the caller to reason about the partial success - this is something we keep coming back to, and that has been the source of numerous bugs in libp2p: aborting operations mid-way often means some sort of cleanup must happen - this is specially true of composite operations and it is exceedingly rare for these not to require local error handling and cleanup - to the point that it should be motivated explicitly.

Reply:
Cool